### ISSUE TITLE
Bring back unsafe BLAS/LAPACK speedups

### ISSUE THREAD

**Issue by RMKirkpatrick** [OpenMx/OpenMx#240]
I first brought up this proposal in a comment to issue #231 :

> As of 8/15/19, @jpritikin replaced all Fortran BLAS/LAPACK calls in the OpenMx backend with equivalent code that uses Eigen. The OpenMx backend used to have an omxBLAS.f, which contained "unsafe" implementations of the BLAS/LAPACK subroutines we use. Those subroutines are "unsafe" because, contrary to IEEE standard, they don't necessarily propagate NaNs and other non-finite values through multiplication. Their lack of "safety", however, made them faster in some circumstances.

> We scrapped src/omxBLAS.f because CRAN required that we stop passing strings to Fortran subroutines. But that change broke something subtle with omxBLAS.f, which Joshua couldn't figure out. Since our backend now no longer uses any calls to Fortran BLAS/LAPACK subroutines, I suggest we re-implement the "unsafety" in Joshua's new Eigen code. By default, it will be switched off, but may be switched on by the user through a runtime option.

It's possible that maybe the unsafe speedups should be implemented as part of the definition of the appropriate MxAlgebra operators instead (or perhaps in addition to?).

**Comment by mcneale** [comment]
Yep, I think this makes sense.  It might even be better to use unsafe by default, as long as optimization fails don't seem much more frequent.
